app/models/ticket_purchase.rb

```diff
@@ -25,18 +25,21 @@ class TicketPurchase < ActiveRecord::Base
 
   def self.purchase(conference, user, purchases)
     errors = []
-    ActiveRecord::Base.transaction do
-      conference.tickets.each do |ticket|
-        quantity = purchases[ticket.id.to_s].to_i
-        # if the user bought the ticket and is still unpaid, just update the quantity
-        purchase = if ticket.bought?(user) && ticket.unpaid?(user)
-                     update_quantity(conference, quantity, ticket, user)
-                   else
-                     purchase_ticket(conference, quantity, ticket, user)
-                   end
-
-        if purchase && !purchase.save
-          errors.push(purchase.errors.full_messages)
+    if count_purchased_registration_tickets(conference, purchases) > 1
```

Comment:
why is not this in a validation?

Reply:
We can not do this in validation because of the way ticket purchase feature has been implemented. Suppose a user try to buy 1 registration ticket of type A and 1 registration ticket of type B in a single purchase. Now 2 ticket purchase object will be created one for type A ticket and other for type B ticket with quantity set as one for both objects. Now looking at the at each object we cannot say it is an invalid ticket purchase object until we know the other ticket purchase objects created alongside it in a single purchase. Therefore we cannot check this in validation level because we need the whole `purchases` object.

Reply:
but you can also add a validation to check that there is not another purchase_ticket with registration ticket, right?

Reply:
Yes. 👍